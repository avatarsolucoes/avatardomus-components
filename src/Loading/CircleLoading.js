import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './loading.css';

export function Carga({ color, duration, size }) {
  const calcW = Math.ceil((size * 15.789) / 100);
  const styleCarga = {
    borderTopColor: color,
    animationDuration: `${duration}s`,
    width: size,
    height: size,
    borderWidth: calcW,
  };
  return <div className={styles.loadingChargeCircle} style={styleCarga} duration={duration} />;
}

Carga.propTypes = {
  color: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
  size: PropTypes.number.isRequired,
};

export default function CircleLoading(props) {
  const { children, text, className, opacity, color, duration, size } = props;

  const returnText = () => {
    if (text) return <div className={styles.loadingText}>{text}</div>;
    return children || null;
  };

  const mainClass = cx(styles.divLoading, className);

  return (
    <div className={mainClass} opacity={opacity}>
      <div className={styles.loadingContent}>
        <div>
          <Carga color={color} duration={duration} size={size} />
          {returnText()}
        </div>
      </div>
    </div>
  );
}

CircleLoading.propTypes = {
  children: PropTypes.any,
  text: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.string,
  duration: PropTypes.number,
  opacity: PropTypes.number,
  size: PropTypes.number,
};

CircleLoading.defaultProps = {
  children: null,
  text: null,
  className: null,
  color: '#000',
  duration: 0.5,
  opacity: 0.5,
  size: 44,
};
