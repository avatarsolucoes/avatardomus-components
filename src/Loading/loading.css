.divLoading {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  align-content: center;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: 100%;
  height: 100%;
  max-height: 100%;
  border: 0;
  background-color: rgba(0, 0, 0, 0.5);
  text-align: center;
}

.divLoading .loadingContent {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  align-content: center;
  border: 0;
  max-width: 100%;
}

.loadingChargeCircle {
  display: block;
  width: 38px;
  height: 38px;
  border: rgba(255, 255, 255, 1) 6px solid;
  border-radius: 50%;
  margin: 16px;
  border-top-color: rgba(0, 0, 0, 1);
  -webkit-animation: loadingRotate linear infinite 0.5s;
  -moz-animation: loadingRotate linear infinite 0.5s;
  animation: loadingRotate linear infinite 0.5s;
}

.divLoading .loadingContent .loadingText {
  display: block;
  max-width: 100%;
  width: 100%;
  text-align: center;
}

@keyframes loadingRotate {
  from {
    -o-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -o-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
