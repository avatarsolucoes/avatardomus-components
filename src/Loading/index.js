import CircleLoading, { Carga } from './CircleLoading';

export { CircleLoading, Carga };
