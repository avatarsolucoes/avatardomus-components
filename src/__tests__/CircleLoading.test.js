import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { CircleLoading } from '../Loading';

describe('CircleLoading', () => {
  it('is truthy', () => {
    expect(CircleLoading).toBeTruthy();
  });

  it('Test component without', () => {
    const wrapper = shallow(<CircleLoading />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Teste renderer whith children', () => {
    const wrapper = renderer
      .create(
        <CircleLoading color="#F00" dutaion={0.7}>
          TESTE
        </CircleLoading>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Teste renderer whith text', () => {
    const wrapper = renderer.create(<CircleLoading color="#F00" dutaion={0.7} text="teste" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
