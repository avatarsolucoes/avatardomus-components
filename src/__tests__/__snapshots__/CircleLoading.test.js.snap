// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLoading Test component without 1`] = `ShallowWrapper {}`;

exports[`CircleLoading Teste renderer whith children 1`] = `
<div
  className=""
  opacity={0.5}
>
  <div>
    <div>
      <div
        duration={0.5}
        style={
          Object {
            "animationDuration": "0.5s",
            "borderTopColor": "#F00",
            "borderWidth": 7,
            "height": 44,
            "width": 44,
          }
        }
      />
      TESTE
    </div>
  </div>
</div>
`;

exports[`CircleLoading Teste renderer whith text 1`] = `
<div
  className=""
  opacity={0.5}
>
  <div>
    <div>
      <div
        duration={0.5}
        style={
          Object {
            "animationDuration": "0.5s",
            "borderTopColor": "#F00",
            "borderWidth": 7,
            "height": 44,
            "width": 44,
          }
        }
      />
      <div>
        teste
      </div>
    </div>
  </div>
</div>
`;
